polynote-spark/src/main/scala/polynote/kernel/remote/RemoteSparkKernel.scala

```diff
@@ -72,8 +72,10 @@ class RemoteSparkKernel(
     */
   private def prepareResultStream(req: RemoteRequest): IO[IO[Stream[IO, Result]]] = for {
     queuePromise <- Deferred[IO, Queue[IO, Option[Result]]]
-    _            <- IO(streamRequests.put(req.reqId, queuePromise))
-  } yield queuePromise.get.map(q => q.dequeue.unNoneTerminate.map(transformResult).onFinalize(IO { streamRequests.remove(req.reqId); () }))
+    semaphore    <- Semaphore[IO](1)
```

Comment:
OK, these changes don't really have anything to do with the `ReprsOf` stuff. Just more tweaking to try and make this reliable enough to pass travis.

This particular round is fixing the synchronization around appending to and closing proxied result streams. It should (hopefully) eliminate the race condition where the stream gets closed before requests finish getting appended.